source/transfer_list.rst

```diff
@@ -657,5 +659,103 @@ such that the last ACPI table in this entry ends at offset
      - hdr_size
      - One or more ACPI tables.
 
+
+.. _tpm_evlog_entry:
+
+TPM event log table entry layout (XFERLIST_EVLOG)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+This entry type holds TPM-related information for a platform. The TPM event log
+info is a region containing a TPM event log as defined by TCG EFI Protocol
+Specification [TCG_EFI]_.
+
+.. _tab_tpm_evlog:
+.. list-table:: TPM event log type layout
+   :widths: 2 2 4 8
+
+   * - Field
+     - Size (bytes)
+     - Offset (bytes)
+     - Description
+
+   * - tag_id
+     - 0x3
+     - 0x0
+     - The tag_id field must be set to **5**.
+
+   * - hdr_size
+     - 0x1
+     - 0x3
+     - |hdr_size_desc|
+
+   * - data_size
+     - 0x4
+     - 0x4
+     - The size of the event log in bytes + sizeof(flags) i.e. 0x4.
+
+   * - flags
+     - 0x4
+     - hdr_size
+     - flags are intended to signal properties of this TE. Bit 0 is
+       need_to_replay flag. Some firmware components may compute measurements
+       to be extended into a TPM and add them to the TPM event log, but those
+       components are unable  to access the TPM themselves. In this case, the
+       component should set the "need_to_replay" flag so that the next
+       component in the boot chain is aware that the TPM has not been updated.
+       A component with access to the TPM would replay the event log by reading
+       each measurement recorded and extending it into the TPM. Once the
+       measurements are extended into the TPM, then the "need_to_replay" flag
+       must be cleared if the transfer list is passed to additional firmware
+       components, means bootloader needs to replay the EventLog. Default
+       value is "0". Other bits should be set to zero.
+   
+   * - event_log
+     - data_size - 0x4
+     - hdr_size + 0x4
+     - Holds a complete event log.
+
+
+.. _tpm_crb_base_entry:
+
+TPM CRB base address table entry layout (XFERLIST_TPM_CRB_BASE)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+The CRB info defines the address of a region of memory that has been carved out
+and reserved for use as a TPM Command Response Buffer interface.
+
+.. _tab_tpm_crb_base:
+.. list-table:: TPM CRB base type layout
+   :widths: 4 2 4 8
+
+   * - Field
+     - Size (bytes)
+     - Offset (bytes)
+     - Description
+
+   * - tag_id
+     - 0x3
+     - 0x0
+     - The tag_id field must be set to **6**.
+
+   * - hdr_size
+     - 0x1
+     - 0x3
+     - |hdr_size_desc|
+
+   * - data_size
+     - 0x4
+     - 0x4
+     - This value should be set to **0xc** i.e. sizeof(crb_base_address) + sizeof(crb_size).
+
+   * - crb_base_address
+     - 0x8
+     - hdr_size
+     - The physical base address of a region of memory reserved for use as a
+       TPM's Command Response Buffer region.
+
+   * - crb_size 
+     - 0x4
+     - hdr_size + 0x8
+     - Size of CRB. If crb_size is "0" that means crb_base_address is not valid.
```

Comment:
If the address is not valid, then this TE should not be provided, right? Or is it useful to describe the size but not where it is? I think this needs some more explanation

Reply:
We should delete "If crb_size is "0" that means crb_base_address is not valid."

That is a vestige of when this was all combined into one transfer list.  If there is no region reserved for the CRB, then the TE will not be present at all.